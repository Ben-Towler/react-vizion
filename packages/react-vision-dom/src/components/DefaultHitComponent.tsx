import { SearchResult } from './HitsComponent';

type Props = {
  searchResult: SearchResult;
};

const DefaultHitComponent: React.FunctionComponent<Props> = ({
  searchResult,
}) => {
  return (
    <div className="default-hit-card">
      <div className="default-hit-container">
        <h4 className="default-hit-title">{searchResult.resourceName}</h4>
        <div className="card-body">{JSON.stringify(searchResult, null, 2)}</div>
      </div>
    </div>
  );
};

export default DefaultHitComponent;
