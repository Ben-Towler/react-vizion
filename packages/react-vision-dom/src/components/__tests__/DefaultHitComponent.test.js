import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DefaultHitComponent from '../DefaultHitComponent';

Enzyme.configure({ adapter: new Adapter() });

describe('DefaultHitComponent', () => {
  const searchResult = { id: '1', resourceName: 'Test', note: 'test' };

  it('applies its default props', () => {
    const instance = renderer.create(
      <DefaultHitComponent searchResult={searchResult} refine={() => null} />
    );

    expect(instance.toJSON()).toMatchSnapshot();

    instance.unmount();
  });

  it('should render with the resource name in the title', () => {
    const wrapper = shallow(
      <DefaultHitComponent searchResult={searchResult} />
    );
    const titleElement = wrapper.find('.default-hit-title');

    expect(titleElement.text()).toEqual(searchResult.resourceName);
  });

  it('should render with the resource as a JSON in the card body', () => {
    const wrapper = shallow(
      <DefaultHitComponent searchResult={searchResult} />
    );
    const cardBodyElement = wrapper.find('.card-body');

    const expectedResult = JSON.stringify(searchResult, null, 2);

    expect(cardBodyElement.text()).toEqual(expectedResult);
  });
});
