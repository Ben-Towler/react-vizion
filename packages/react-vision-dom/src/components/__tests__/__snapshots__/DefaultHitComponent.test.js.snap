// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultHitComponent applies its default props 1`] = `
<div
  className="default-hit-card"
>
  <div
    className="default-hit-container"
  >
    <h4
      className="default-hit-title"
    >
      Test
    </h4>
    <div
      className="card-body"
    >
      {
  "id": "1",
  "resourceName": "Test",
  "note": "test"
}
    </div>
  </div>
</div>
`;
